### DISCUSSION TITLE
Non semver version numbering in Sentry dashboard

### DISCUSSION THREAD

**Discussion by vtwigt** [stayallive/wp-sentry#188] [Q&A]
Even though my theme is using semver notation for version numbers, I found that the Sentry dashboard stated the version numbers as "non-semver":

<img width="632" alt="Screenshot 2024-05-16 at 16 18 38" src="https://github.com/stayallive/wp-sentry/assets/135724179/5558406d-3268-4f2f-9e00-bc38702229d7">

The plugin sets the `release` option with the exact theme version (i.e. 2.6.3).
However, it looks like the [Sentry docs](https://docs.sentry.io/platforms/javascript/configuration/releases/#setting-a-release) mention a format like `my-project-name@2.3.12`.

Could this be the reason that Sentry does not recognize the semver format used?
And if so, could this be fixed? For example, by using the theme folder name: `theme-folder@2.6.3`

**Comment by stayallive** [comment]
I am not a 100% sure that is it, can you check your previous releases? According to the docs this is how Sentry decides if a release strategy for a project is semver:

![image](https://github.com/stayallive/wp-sentry/assets/1090754/8d7168af-e504-46e6-9a0a-206fc4b7ed8f)

I don;t think the package name prefix is required but I will ask around to find out!

**Reply by vtwigt** [reply]
This project has more than 10 releases, so it should fall under the third category you mentioned. Still, Sentry seems to mark all releases as non-semver:
<img width="623" alt="Screenshot 2024-05-30 at 16 45 37" src="https://github.com/stayallive/wp-sentry/assets/135724179/0632961c-5227-4afa-876f-072417d60976">

**Reply by cristiano2lopes** [reply]
I am seeing the same thing. with a slightly different scheme.
![2025-03-06_180847@2x](https://github.com/user-attachments/assets/34d47073-3d9a-4e4d-b2c6-a319ba0f83bd)

**Reply by stayallive** [reply]
If I am reading the code correctly you do need the prefix in your release: https://github.com/getsentry/sentry/blob/e981f3345b002ad3ccef31cc5a3b366e53f70616/src/sentry/models/release.py#L333-L351

So a server version looks like: `my.project.name@2.3.12+1234` for example, see: https://docs.sentry.io/platforms/javascript/configuration/releases/#bind-the-version

**Reply by vtwigt** [reply]
> So a server version looks like: `my.project.name@2.3.12+1234` for example, see: https://docs.sentry.io/platforms/javascript/configuration/releases/#bind-the-version

@stayallive , so that would imply that, to get semantic versioning to be detected properly, the WordPress Sentry plugin should include a unique identifier as `package` name, correct?

May I suggest to use the theme name or text domain as `package` by default?
So that would result in sending `themename@2.3.12+1234` or simply `themename@2.3.12` to Sentry.

**Reply by stayallive** [reply]
This version is not set by yourself but automatically by the package?

**Reply by vtwigt** [reply]
I'm using the default behavior, as explained in the [docs](https://github.com/stayallive/wp-sentry/tree/v8.3.1#wp_sentry_version):
> Define a version of your site. By default the active theme version will be used or unspecified if theme version could not be resolved.

So this means that by default the plugin only uses the theme version, but not the name, resulting in a string that is not adhering Sentry's requirements for semver support.

Hence my suggestion to add the theme folder name, name or language identifier as a prefix to the default version string that the plugin generates, so that compatibility with Sentry's semver implementation is ensured.
